### ISSUE TITLE
2021-005 - SMDS - Annual Report Updates

### ISSUE THREAD

**Issue by anthonyperk** [CurrentlyNotSet/SERB#752]
*****45 Entries to change****

16 - HEARINGS HELD (Correct) Qty 8
Query: (Count of Returned Results)
SELECT * FROM HearingCase WHERE hearingdate BETWEEN '07/01/2020' AND '06/30/2021'

17 - BOARD MEETINGS (Correct) Qty 16 + 1 Emergency = 17
Query: (Count of Unique Board Meeting Dates)
SELECT * FROM BoardMeeting where boardMeetingDate BETWEEN '2020-07-01' AND '2021-06-30'

24 – REP/ULP CASES MEDIATED (Correct) Qty 12
Query:
SELECT * FROM HearingsMediation WHERE MediationDate BETWEEN '07/01/2020' AND
'06/30/2021'
Note: Use unique cases

25 – REP/ULP MEDIATION MEETINGS SESSIONS (Correct) Qty 12 (due to virtual there was actually 17
session)
Query:
SELECT * FROM HearingsMediation WHERE MediationDate BETWEEN '07/01/2020' AND
'06/30/2021'
Note: Use all cases; duplicate cases indicate multi-day

26 – REP/ULP SETTLEMENTS (Correct) Qty 8
Query:
SELECT * FROM HearingsMediation WHERE MediationDate BETWEEN '07/01/2020' AND
'06/30/2021' AND outcome = 'Settled'

30 – RECOGNITION REQUESTS/ELECTION PETITIONS CERTIFIED (Correct) Qty 14
Query: (Update existing query)
SELECT CaseYear, CaseType, CaseMonth, CaseNumber, pollingenddate, finalboarddate,
boardActionType, boardActionDate FROM REPCase WHERE boardactiondate BETWEEN '07/01/2020'
AND '06/30/2021' AND (BoardActionType = 'Cert VR' OR BoardActionType = 'Cert Opt-IN VR')

31 – RECOGNITION REQUESTS – ELECTION PETITIONS DISMISSED (Correct) Qty 29
Query: (Update existing query)
SELECT CaseYear, CaseType, CaseMonth, CaseNumber, pollingenddate, finalboarddate,
boardActionType, boardActionDate, professionalNonProfessional, professionalIncluded,
professionalExcluded, nonProfessionalIncluded, nonProfessionalExcluded FROM REPCase WHERE
finalBoardDate BETWEEN '07/01/2020' AND '06/30/2021' AND (BoardactionType = 'Dism AC' or
BoardactionType = 'Dism Opt-IN RC' or BoardactionType = 'Dism Opt-IN VR' OR BoardactionType =
'Dism RC' OR BoardactionType = 'Dism RD' OR BoardactionType = 'Dism VR')

32 - MISCELLANEOUS REPRESENTATION ACTIVITES (Correct) Qty 53
Query: (Update existing query)
SELECT CaseYear, CaseType, CaseMonth, CaseNumber, pollingenddate, finalboarddate,
boardActionType, boardActionDate, professionalNonProfessional, professionalIncluded,
professionalExcluded, nonProfessionalIncluded, nonProfessionalExcluded FROM REPCase WHERE
finalBoardDate BETWEEN '07/01/2020' AND '06/30/2021' AND (BoardactionType = 'Appr AC' OR
BoardactionType = 'Appr JTAC' or BoardactionType = 'Appr UC' or BoardactionType = 'Approve' OR
BoardactionType = 'Brd Det Unit' OR BoardactionType = 'Cert Opt-In RC' OR BoardactionType = 'DIR
ELECT' OR BoardactionType = 'Dir Evid Hrg' OR BoardactionType = 'Dir Hrg' OR BoardactionType =
'Dir Inq' OR BoardactionType = 'DIR MAIL ELECT' OR BoardactionType = 'Dir Med Hrg' OR
BoardactionType = 'Dir Non-Oral Hrg' OR BoardactionType = 'Dir Oral Hrg' OR BoardactionType = 'Dir
Pre-D Med' OR BoardactionType = 'DIR RUNOFF' OR BoardactionType = 'Dism' OR BoardactionType
= 'Dism AC' OR BoardactionType = 'Dism JTAC' OR BoardactionType = 'DISM OBJ CERT RES' OR
BoardactionType = 'Dism UC' OR BoardactionType = 'REVOKE' OR BoardactionType = 'Stay' OR
BoardactionType = 'Postponed' OR BoardactionType = 'Remand' OR BoardactionType = 'Table')

55 – CONTRACTS MEDIATED (Could be tracked once REP/ULP/Mediation Tab built; count of entries
within date range)
Query:
Manual… unable to track in system.

56 – MEDIATION MEETINGS/SESSIONS HELD (Could be tracked once REP/ULP Mediation Tab built;
count of entries within date range)
Query:
Manual… unable to track in system.

57 – SETTLEMENTS (Could be tracked once REP/ULP/Mediation Tab built; count of entries within
date range)
Query:
Manual… unable to track in system.

85 – PUBLIC SECTOR STRIKES – EDUCATION (Correct) Qty 1
Query: (Update existing query)
SELECT CaseYear, CaseType, CaseMonth, CaseNumber, FileDate, StrikeFileDate,
noticeOfIntentToStrikeOnly, noticeOfIntentToStrikeAndPicket, FFEmployerType,FFEmployeeType,
strikeBegan, strikeEnded, totalNumberOfDays FROM MEDCase WHERE CaseType = 'STK' and
FFEmployerType = 'School District'and strikeBegan BETWEEN '07/01/2020' AND '06/30/2021'

86 – PUBLIC SECTOR STRIKES – CITY (Correct) Qty 0
Query: (Update existing query)
SELECT CaseYear, CaseType, CaseMonth, CaseNumber, FileDate, StrikeFileDate,
noticeOfIntentToStrikeOnly, noticeOfIntentToStrikeAndPicket, FFEmployerType,FFEmployeeType,
strikeBegan, strikeEnded, totalNumberOfDays FROM MEDCase WHERE CaseType = 'STK' and
FFEmployerType = 'City'and strikeBegan BETWEEN '07/01/2020' AND '06/30/2021'

87 – PUBLIC SECTOR STRIKES – COUNTY (Correct) Qty 0
Query: (Update existing query)
SELECT CaseYear, CaseType, CaseMonth, CaseNumber, FileDate, StrikeFileDate,
noticeOfIntentToStrikeOnly, noticeOfIntentToStrikeAndPicket, FFEmployerType,FFEmployeeType,
strikeBegan, strikeEnded, totalNumberOfDays FROM MEDCase WHERE CaseType = 'STK' and
FFEmployerType = 'County'and strikeBegan BETWEEN '07/01/2020' AND '06/30/2021'

88 – PUBLIC SECTOR STRIKES – TOWNSHIP (Correct) Qty 0
Query: (Update existing query)
SELECT CaseYear, CaseType, CaseMonth, CaseNumber, FileDate, StrikeFileDate,
noticeOfIntentToStrikeOnly, noticeOfIntentToStrikeAndPicket, FFEmployerType,FFEmployeeType,
strikeBegan, strikeEnded, totalNumberOfDays FROM MEDCase WHERE CaseType = 'STK' and
FFEmployerType = 'Township'and strikeBegan BETWEEN '07/01/2020' AND '06/30/2021'

89 – PUBLIC SECTOR STRIKES – OTHER (Correct) Qty 1
Query: (Update existing query)
SELECT CaseYear, CaseType, CaseMonth, CaseNumber, FileDate, StrikeFileDate,
noticeOfIntentToStrikeOnly, noticeOfIntentToStrikeAndPicket, FFEmployerType,FFEmployeeType,
strikeBegan, strikeEnded, totalNumberOfDays FROM MEDCase WHERE CaseType = 'STK' AND
(FFEmployerType = 'Other' or FFEmployerType = 'University' or FFEmployerType = 'State Govt.') and
strikeBegan BETWEEN '07/01/2020' AND '06/30/2021'

90 – PUBLIC SECTOR STRIKES – TOTAL (Correct) Qty 2
Query: (Update existing query)
SELECT CaseYear, CaseType, CaseMonth, CaseNumber, FileDate, StrikeFileDate,
noticeOfIntentToStrikeOnly, noticeOfIntentToStrikeAndPicket, FFEmployerType,FFEmployeeType,
strikeBegan, strikeEnded, totalNumberOfDays FROM MEDCase WHERE CaseType = 'STK' and
strikeBegan BETWEEN '07/01/2020' AND '06/30/2021'

91 – ELECTIONS HELD (Correct) Qty 53 + 2 Runoffs = 55
Query: (Update existing Query)
SELECT CaseYear, CaseType, CaseMonth, CaseNumber, pollingenddate, finalboarddate,
boardActionType, boardActionDate, professionalNonProfessional, professionalIncluded,
professionalExcluded, nonProfessionalIncluded, nonProfessionalExcluded FROM REPCase WHERE
pollingenddate BETWEEN '07/01/2020' AND '06/30/2021' AND BoardActionType = 'CERT RESULTS'

93 – CHOICES FOR REPRESENTATION (Correct) Qty 82 Query pulls correct cases, need to perform
the column counts excluding where field = No Representative
Query: (Update existing query)
(NOTE: Count 1 when BallotOne, BallotTwo, BallotThree, BallotFour contain data other than “No
Representation”)
SELECT CaseYear, CaseType, CaseMonth, CaseNumber, pollingenddate, finalboarddate,
boardActionType, boardActionDate, professionalNonProfessional, professionalIncluded,
professionalExcluded, nonProfessionalIncluded, nonProfessionalExcluded, ballotone, ballottwo,
ballotthree, ballotfour FROM REPCase WHERE pollingenddate BETWEEN '07/01/2020' AND
'06/30/2021'

97 – REP CASES ADJUDICATED (ALL Board Actions) (Correct) = Quantity 200 + 9
Query: (Add count of two queries below together)
Query 1:
SELECT * FROM BoardMeeting WHERE BoardMeetingdate BETWEEN '07/01/2020' AND '06/30/2021'
AND CaseType = 'REP'
Query 2:
SELECT CaseYear, CaseType, CaseMonth, CaseNumber FROM HearingCase WHERE BoardActionDate
BETWEEN '07/01/2020' AND '06/30/2021' AND CaseType = 'REP'

98 – AVERAGE REP CASE PROCESSING TIME Note: Quantity should = 100 days for July 1, 2020 – June 30, 202
Query: Note: Compute the average between the Final Board Date and the File Date
SELECT CaseYear, CaseType, CaseMonth,CaseNumber, Status1, Status2, Filedate, finalBoardDate,
dateofappeal, courtcloseddate, boardactiontype, boardactiondate FROM REPCase WHERE
FinalBoardDate BETWEEN '07/01/2020' AND '06/30/2021' (Case Selection… does not include the
math)
Note: The math is the number of days from File date to Tally Date (or final Memo date if no Tally)

99 – MEDIATIONS CONDUCTED BY REP INVESTIGATORS (PRE-PC) (Could be tracked once
REP/ULP/Mediation Tab built; count of entries within date range)
Query:
Manual… unable to track in system.

107 – ULP CASES ADJUDICATED (ALL BOARD ACTIONS) (Correct) = Quantity 278 + 16 = 294
Query: (Add two queries below together)
Query 1:
SELECT * FROM BoardMeeting WHERE BoardMeetingdate BETWEEN '07/01/2020' AND '06/30/2021'
AND CaseType = 'ULP'
Query 2:
SELECT CaseYear, CaseType, CaseMonth, CaseNumber FROM HearingCase WHERE BoardActionDate
BETWEEN '07/01/2020' AND '06/30/2021' AND CaseType = 'ULP'

108 – AVERAGE ULP PROCESSING TIME (Could be tracked once ULP enhancement in place
Query:
Manual… unable to track in system.

109 – MEDIATIONS CONDUCTED BY ULP INVESTIGATORS(Could be tracked once REP/ULP/Mediation
Tab built; count of entries within date range)
Query:
Manual… unable to track in system.

110 – TOTAL ALLEGATIONS OF VIOLATIONS (Correct) = 233 (includes A, B, D and ERC allegations)
Query: (Total of four rows below – update existing query)
SELECT CaseYear, CaseType, CaseMonth, CaseNumber, Filedate FROM ULPCase WHERE (CaseType =
'ULP' OR CaseType = 'ERC' OR CaseType = 'JWD') and filedate BETWEEN '07/01/2020' AND
'06/30/2021'

113 – SECTION 4117.11(D) ALLEGED JWD (Correct) Qty = 0
Query:
SELECT CaseYear, CaseType, CaseMonth, CaseNumber,allegation, Filedate FROM ULPCase WHERE
CaseType = 'JWD' and filedate BETWEEN '07/01/2020' AND '06/30/2021'

114 – ALLEGED ERC (Correct) Qty = 1
Query:
SELECT CaseYear, CaseType, CaseMonth, CaseNumber,allegation, Filedate FROM ULPCase WHERE
CaseType = 'ERC' and filedate BETWEEN '07/01/2020' AND '06/30/2021'

115 – BOARD FINDING OF STATUTORY VIOLATIONS (Correct) Qty = 0
Query: (Update existing query)
SELECT * FROM HearingCase WHERE issuanceOfOptionOrDirectiveDate BETWEEN '07/01/2020' AND
'06/30/2021' AND (CaseType = 'ULP' OR CaseType = 'JWD' OR CaseType = 'ERC') and finalResult =
'Adjudicated - Violation'

116 – ALJPOs/ALJRDs (Correct) Qty = 7
Query: (Update Existing Query)
SELECT * FROM HearingCase WHERE proposedRecIssuedDate BETWEEN '07/01/2020' AND
'06/30/2021' AND ALJID <> '1464' and ALJID <>'1449' AND ALJID <> '1468' (Excluding General
Counsel, Staff Attorney, Full Board)

117 – PRETRIALS HELD (Correct) Qty = 3
Query: (Update Existing Query)
SELECT * FROM HearingCase WHERE preHearingDate BETWEEN '07/01/2020' AND '06/30/2021' AND
ALJID <> '1464' and ALJID <>'1449' AND ALJID <> '1468' (Excluding General Counsel, Staff Attorney,
Full Board)

120 – GENERAL COUNSEL – REP INQUIRIES – R&R ISSUED (Correct) Qty = 0
Query:
SELECT * FROM HearingCase WHERE proposedRecIssuedDate BETWEEN '07/01/2020' AND
'06/30/2021' AND CaseType = 'REP' AND ALJID ='1468' (Michael Allen)

121 – GENERAL COUNSEL – ULP HEARING – PO ISSUED (Correct) Qty = 0
Query:
SELECT * FROM HearingCase WHERE proposedRecIssuedDate BETWEEN '07/01/2019' AND
'06/30/2020' AND CaseType = 'ULP' AND ALJID ='1468' (Michael Allen)
122 – STAFF ATTORNEY – REP INQUIRIES – R&R ISSUED (Correct) Qty = 2
Query:
SELECT * FROM HearingCase WHERE proposedRecIssuedDate BETWEEN '07/01/2020' AND
'06/30/2021' AND CaseType = 'REP' AND ALJID = '1464' (Robert Fekete)

123 – STAFF ATTORNEY – ULP HEARING – PO ISSUED (Correct) Qty = 0
Query:
SELECT * FROM HearingCase WHERE proposedRecIssuedDate BETWEEN '07/01/2020' AND
'06/30/2021' AND CaseType = 'ULP' AND ALJID ='1464' (Robert Fekete)

127 – NUMBER OF PRE-HEARINGS (CASE COUNT) – SPBR Query:
Manual… unable to track in system due to inconsistent data entry on Activity Tab and Hearing Tab

128 – DAYS OF PRE-HEARING
Query:
Manual… unable to track in system due to inconsistent data entry on Activty Tab

129 – NUMBER OF HEARINGS (CASE COUNT) – SPBR
Query:
Manual… unable to track in system due to inconsistent data entry on Activity Tab and Hearing Tab
SELECT * FROM CMDSHearing WHERE HearingType = 'RH' and hearingdatetime BETWEEN '07/01/2020'
AND '06/30/2021'

130 – Days of Hearing – SPBR
Query:
Manual… unable to track in system due to inconsistent data entry on activity tab

131 – CASES RESOLVED – SPBR (Correct) Qty = 355
Query:
SELECT * FROM CMDSCase WHERE closedate BETWEEN '07/01/2020' AND '06/30/2021'

132 – NUMBER OF PRE-HEARINGS (CASE COUNT) – SERB (Accurate Qty 4)
Query:
SELECT * FROM HearingCase where prehearingdate BETWEEN '07/01/2020' AND '06/30/2021'

133 – DAYS OF PRE-HEARING
Manual… unable to track in system due to inconsistent data entry on activity tab

134 – NUMBER OF HEARINGS (CASE COUNT) – SERB (Accurate Qty 8)
Query:
SELECT * FROM HearingCase where hearingdate BETWEEN '07/01/2020' AND '06/30/2021'

135 – DAYS OF HEARING
Manual… unable to track in system due to inconsistent data entry on activity tab

136 – CASES RESOLVED – SERB
Manual… from Erin

![image](https://user-images.githubusercontent.com/24595073/131170828-8916ef23-9858-438e-9ddd-610efc034b95.png)

![image](https://user-images.githubusercontent.com/24595073/131170840-5da88405-8791-460f-8aa0-eab62218d0a2.png)

![image](https://user-images.githubusercontent.com/24595073/131170851-76a153f5-a6ed-47e3-9085-807f4e9f321b.png)

![image](https://user-images.githubusercontent.com/24595073/131170858-3b90fbad-868f-4857-a711-d88f45d80785.png)

**Comment by anthonyperk** [comment]
I have attached a txt file that formatted the changes to the queries better than copying them here

There are 13 new bookmarks (oneonesix   through   onetwoeight)

3 of the queries are weird and I am not sure how to format them to work in the jave code OR have no idea how to write the query:

eighteight (existing bookmark) - The query I wrote works correctly, not sure how to format it into the java code

oneoneseven (new bookmark) - This technically combines two totals into one bookmark.  When I tried to format the queries into one query, I couldnt get the java formatting correct with the '+' symbol actually in the query itself so I divided them into two bookmarks then added the bookmarks together.  Not sure if that will work.  I added the combined query below the java queries.

oneoneeight  (new bookmark) - I have tired several way to get this to work, never works.  I am probably just dumb
[2021-005 Annual Report Queries.txt](https://github.com/CurrentlyNotSet/SERB/files/7781220/2021-005.Annual.Report.Queries.txt)

**Comment by anthonyperk** [comment]
I have committed these changes and am waiting on Andrew to check

**Comment by altrdgenetics** [comment]
I will need help with this. new one

Are you looking for me to loop through each row and take the date difference and then total up all of the date differences and THEN divide that by the total number of cases?

**Comment by anthonyperk** [comment]
UPDATED "Total Number of Days (File Date -> Final Board Date)" query from above:

SELECT DISTINCT
ULPCase.caseYear + '-' + ULPCase.caseType + '-' + ULPCase.caseMonth + '-' + ULPCase.caseNumber AS CaseNumber,
(SELECT TOP 1 BoardMeeting.boardmeetingdate FROM BoardMeeting WHERE BoardMeeting.caseYear = ULPCase.caseYear AND BoardMeeting.caseType = ULPCase.caseType AND BoardMeeting.caseMonth = caseMonth AND BoardMeeting.caseNumber = caseNumber ORDER BY boardMeetingDate DESC) AS FinalBoardMeetingDate,
ULPCase.fileDate,
DATEDIFF(dd, ULPCase.fileDate, (SELECT TOP 1 BoardMeeting.boardmeetingdate FROM BoardMeeting WHERE BoardMeeting.caseYear = ULPCase.caseYear AND BoardMeeting.caseType = ULPCase.caseType AND BoardMeeting.caseMonth = caseMonth AND BoardMeeting.caseNumber = caseNumber ORDER BY boardMeetingDate DESC) ) AS DIFF


FROM ULPCase
INNER JOIN BoardMeeting ON
ULPCase.caseYear = BoardMeeting.caseYear AND
ULPCase.caseType = BoardMeeting.caseType AND
ULPCase.caseMonth = BoardMeeting.caseMonth AND
ULPCase.caseNumber = BoardMeeting.caseNumber

WHERE ULPCase.finalDispositionStatus IS NOT NULL AND (BoardMeeting.boardMeetingDate BETWEEN '07/01/2020'AND '06/30/2021')






> Are you looking for me to loop through each row and take the date difference and then total up all of the date differences and THEN divide that by the total number of cases?

Correct.  The First query will return X amount of rows with Case Number, FinalBoardMeetingDate, FileDate, and DIFF in each row.  If you could add up all of the entries in the 'DIFF' column into one number, then divide it into the number of rows in that query OR use the second query above ("Number of CASES").
